### PR TITLE
changefeed(ticdc): refactor new changefeed method attach the reactor directly

### LINKED ISSUE #6761: refactor new changefeed method

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #6761 

### What is changed and how it works?

refactor `newChangefeed` method, attach the changefeed reactor directly, and simplify the related methods' signatures.

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -125,9 +125,14 @@ type changefeed struct {
 	newScheduler func(ctx cdcContext.Context, startTs uint64) (scheduler.Scheduler, error)
 }
 
-func newChangefeed(id model.ChangeFeedID, up *upstream.Upstream) *changefeed {
+func newChangefeed(
+	id model.ChangeFeedID,
+	state *orchestrator.ChangefeedReactorState,
```
Comment (asddongmen) [inline cdc/owner/changefeed.go:130]: please add a comment in changefeed struct to tell `state` is a read-only fileld.

**Review by asddongmen — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**